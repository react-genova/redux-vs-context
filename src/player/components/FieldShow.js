import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const FieldShow = ({ label, value, className }) => {
    console.log('render FieldShow', label, value);
    return <span className={className}>{`${label}: ${value}`}</span>;
};

FieldShow.propTypes = {
    className: PropTypes.string,
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
};

const FieldShowStyled = styled(FieldShow)`
    margin: 5px;
    border: 1px solid green;
    background-color: #AFA;
    padding: 10px;
`;

export default FieldShowStyled;
