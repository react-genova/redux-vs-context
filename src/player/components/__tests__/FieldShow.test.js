import React from 'react';
import FieldShow from '../FieldShow';
import { render } from 'react-testing-library';
import 'jest-styled-components';

describe('FieldShow component', () => {
    global.console = { log: () => null };

    it('renders without crashing', () => {
        const { container } = render(
            <FieldShow value="the value" label="the label" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
