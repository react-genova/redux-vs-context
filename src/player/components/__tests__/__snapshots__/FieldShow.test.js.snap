// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldShow component renders without crashing 1`] = `
.c0 {
  margin: 5px;
  border: 1px solid green;
  background-color: #afa;
  padding: 10px;
}

<span
  class="c0"
>
  the label: the value
</span>
`;
